src/lines_to_params.rs

```diff
@@ -0,0 +1,165 @@
+use crate::parser;
+use pyo3::prelude::*;
+use pyo3::types::PyDict;
+
+type ActionEffectParams = (u8, u8, u8, u8, u16, u8, u8);
+type StatusEffectParams = (u16, u16, f32, u32);
+
+type StatuslistParams<'a> = (
+    (u32, &'a str),
+    &'a str,
+    Vec<u32>,
+    Vec<f32>,
+    Vec<(u16, u16, f32, u32)>,
+);
+
+/// Get id name type from tuple
+#[pyfunction]
+#[pyo3(text_signature = "(name_id_pair: list[str]) -> (int, str)")]
+pub(crate) fn parse_id_name_pair(inp: Vec<&str>) -> (u32, &str) {
+    (
+        parser::u32_from_param(inp.first().unwrap()),
+        inp.last().unwrap(),
+    )
+}
+
+/// Params to ability
+#[pyfunction]
+#[pyo3(text_signature = "(timestamp: int, params: list[str]) -> Ability")]
+pub(crate) fn ability_from_params(timestamp: i64, inp: Vec<&str>) -> PyObject {
+    let mut col = inp;
+    let source_actor = col.drain(..2).collect::<Vec<&str>>();
+    let ability = col.drain(..2).collect::<Vec<&str>>();
+    let target_actor = col.drain(..2).collect::<Vec<&str>>();
+    let action_effects = col.drain(..16).collect::<Vec<&str>>();
+    let source_resources = col.drain(..6)
+        .map(|x| parser::u32_from_str(x))
+        .collect::<Vec<u32>>();
+    let source_position = col.drain(..4)
+        .map(|x| parser::f32_from_str(x))
+        .collect::<Vec<f32>>();
+    let target_resources = col.drain(..6)
+        .map(|x| parser::u32_from_str(x))
+        .collect::<Vec<u32>>();
+    let target_position = col.drain(..4)
+        .map(|x| parser::f32_from_str(x))
+        .collect::<Vec<f32>>();
+    let sequence = parser::u32_from_param(col.first().unwrap());
+    Python::with_gil(|py| {
```

Comment:
Is the GIL an actual issue here?

Reply:
The GIL has to be activated here due to me creating python objects inside of rust and this is the only way to do this.

Reply:
Sure. The question was more, why? I don't see what you're getting out of the python calls here other than overhead and invalidating some of the performance gains.

Reply:
Continue this in the conversation underneath.